azurerm/internal/services/web/app_service_managed_certificate_resource.go

```diff
@@ -0,0 +1,246 @@
+		}
+	}
+
+	certificate := web.Certificate{
+		CertificateProperties: &web.CertificateProperties{
+			CanonicalName: utils.String(customHostnameBindingId.Name),
+			ServerFarmID:  utils.String(appServicePlanID),
+			Password:      new(string),
+		},
+		Location: utils.String(appServiceLocation),
+		Tags:     tags.Expand(t),
+	}
+
+	if resp, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.CertificateName, certificate); err != nil {
+		// API returns 202 where 200 is expected - https://github.com/Azure/azure-sdk-for-go/issues/13665
+		if !utils.ResponseWasStatusCode(resp.Response, 202) {
+			return fmt.Errorf("Error creating/updating App Service Managed Certificate %q (Resource Group %q): %s", id.CertificateName, id.ResourceGroup, err)
+		}
+	}
+
+	certificateWait := &resource.StateChangeConf{
+		Pending:    []string{"NotFound", "Unknown"},
+		Target:     []string{"Success"},
+		MinTimeout: 1 * time.Minute,
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Refresh: func() (interface{}, string, error) {
+			resp, err := client.Get(ctx, id.ResourceGroup, id.CertificateName)
+			if err != nil {
+				if utils.ResponseWasNotFound(resp.Response) {
+					return "NotFound", "NotFound", nil
+				}
+				return "Unknown", "Unknown", err
+			}
+			if utils.ResponseWasStatusCode(resp.Response, 200) {
+				return "Success", "Success", nil
+			}
+			return "Unknown", "Unknown", err
+		},
+	}
+
+	if !d.IsNewResource() {
+		certificateWait.Timeout = d.Timeout(schema.TimeoutUpdate)
+	}
+
+	if _, err := certificateWait.WaitForState(); err != nil {
+		return fmt.Errorf("waiting for App Service Managed Certificate %q: %+v", id.CertificateName, err)
+	}
+
+	d.SetId(id.ID(""))
+
+	return resourceArmAppServiceManagedCertificateRead(d, meta)
+}
+
+func resourceArmAppServiceManagedCertificateRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Web.CertificatesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ManagedCertificateID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.CertificateName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] App Service Managed Certificate %q (Resource Group %q) was not found - removing from state", id.CertificateName, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on App Service Managed Certificate %q (Resource Group %q): %+v", id.CertificateName, id.ResourceGroup, err)
+	}
+
+	if props := resp.CertificateProperties; props != nil {
+		d.Set("canonical_name", props.CanonicalName)
+		d.Set("friendly_name", props.FriendlyName)
+		d.Set("subject_name", props.SubjectName)
+		d.Set("host_names", props.HostNames)
+		d.Set("issuer", props.Issuer)
+		d.Set("issue_date", props.IssueDate.Format(time.RFC3339))
+		d.Set("expiration_date", props.ExpirationDate.Format(time.RFC3339))
```

Comment:
there's two potential crashes here if `IssueDate` or `ExpirationDate` aren't returned for whatever reason, which is unlikely but we should still catch:

```suggestion
		expirationDate := ""
		if props.ExpirationDate != nil {
			expirationDate = props.ExpirationDate.Format(time.RFC3339)
		}
		d.Set("expiration_date", expirationDate)
```